### PR TITLE
Add number literals

### PR DESCRIPTION
**Pull request by Glyphack**
Closes: #13
Closes: #14

### CODE REVIEW

**Comment by Glyphack** [comment]
Please double check test cases

File: src/snapshots/enderpy__lexer__tests__snapshot_test_lexer-83.snap [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+source: src/lexer.rs
+---
```
Comment (frustak) [inline src/snapshots/enderpy__lexer__tests__snapshot_test_lexer-83.snap:3]: Why don't octal test cases have any description like others?

**Review by frustak — APPROVED** [review]

File: src/snapshots/enderpy__lexer__tests__snapshot_test_lexer-83.snap (hunk 1 above)
Comment (Glyphack) [reply src/snapshots/enderpy__lexer__tests__snapshot_test_lexer-83.snap:3]: I don't know 😂

Comment (frustak) [reply src/snapshots/enderpy__lexer__tests__snapshot_test_lexer-83.snap:3]: yes XD

**Pull request merged**